src/types/form.ts

```diff
@@ -256,6 +257,25 @@ export interface IFormDocument extends IFormSchema {
   responseMode: NonNullable<IFormSchema['responseMode']>
 }
 
+export interface IEncryptedFormDocument extends IEncryptedFormSchema {
+  form_fields: NonNullable<IEncryptedFormSchema['form_fields']>
+  form_logics: NonNullable<IEncryptedFormSchema['form_logics']>
+  permissionList: NonNullable<IEncryptedFormSchema['permissionList']>
+  hasCaptcha: NonNullable<IEncryptedFormSchema['hasCaptcha']>
+  authType: NonNullable<IEncryptedFormSchema['authType']>
+  status: NonNullable<IEncryptedFormSchema['status']>
+  inactiveMessage: NonNullable<IEncryptedFormSchema['inactiveMessage']>
+  // NOTE: Due to the way creating a form works, creating a form without specifying submissionLimit will throw an error.
+  // Hence, using Exclude here over NonNullable.
+  submissionLimit: Exclude<IEncryptedFormSchema['submissionLimit'], undefined>
+  isListed: NonNullable<IEncryptedFormSchema['isListed']>
+  startPage: Required<NonNullable<IEncryptedFormSchema['startPage']>>
+  endPage: Required<NonNullable<IEncryptedFormSchema['endPage']>>
+  webhook: Required<NonNullable<IEncryptedFormSchema['webhook']>>
+  responseMode: NonNullable<IEncryptedFormSchema['responseMode']>
+  publickey: NonNullable<IEncryptedFormSchema['publicKey']>
+}
+
```

Comment:
extract to generic type, as discussed

Reply:
[ddefa69](https://github.com/opengovsg/FormSG/pull/5929/commits/ddefa69b3ec11e1dd258ae77f49ada250b8d45d4)